plugins/unicodelib/unicodelib.cpp

```diff
@@ -718,6 +719,48 @@ unsigned unicodeEditDistanceV4(UnicodeString & left, UnicodeString & right, unsi
     return da[mask(leftLen-1)][rightLen-1];
 }
 
+UnicodeString excludeNthWord(RuleBasedBreakIterator& bi, UnicodeString const & source, unsigned n)
+{
+	UnicodeString processed;
+	processed.append(source);
+	bi.setText(source);
+	int32_t idx = bi.first();
+	int32_t wordidx = 0; //counts the # of words through the iteration of the below while loop.
+	while (idx != BreakIterator::DONE) {
+		int breakType = bi.getRuleStatus();
+		if (breakType != UBRK_WORD_NONE) {
+			// Exclude spaces, punctuation, and the like.
+			//   A status value UBRK_WORD_NONE indicates that the boundary does
+			//   not start a word or number.
+			if (++wordidx == n) {
+				unsigned wordBeginning;
+				if (n == 1) {
+					wordBeginning = 0;
+				}
+				else {
+					wordBeginning = bi.preceding(idx);
```

Comment:
By this do you mean set wordBeginning's value to the idx before the nth word? @ghalliday